src/Converter.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+namespace Money;
+
+final class Converter
+{
+    /**
+     * @var Currencies
+     */
+    private $currencies;
+
+    /**
+     * @param Currencies $currencies
+     */
+    public function __construct(Currencies $currencies)
+    {
+        $this->currencies = $currencies;
+    }
+
+    /**
+     * @param Money        $money
+     * @param CurrencyPair $currencyPair
+     * @param int          $roundingMode
+     *
+     * @return Money
+     */
+    public function convert(Money $money, CurrencyPair $currencyPair, $roundingMode = Money::ROUND_HALF_UP)
+    {
+        if ($money->getCurrency()->getCode() !== $currencyPair->getBaseCurrency()->getCode()) {
+            throw new \InvalidArgumentException('Base Currency of currency pair does not equal money currency');
+        }
+
+        $subunitBaseCurrency = $this->currencies->subunitFor($currencyPair->getBaseCurrency());
+        $subunitCounterCurrency = $this->currencies->subunitFor($currencyPair->getCounterCurrency());
+        $differenceInSubunit = $subunitBaseCurrency - $subunitCounterCurrency;
+        $ratio = $currencyPair->getConversionRatio();
+
+        if ($differenceInSubunit > 0) {
```

Comment:
The following seems to be the same instead of two ifs:

``` php
        if (0 !== $differenceInSubunit) {
            $ratio = $ratio * pow(10, abs($differenceInSubunit));
        }
```

Even if there is a need to have two separate checks: the difference cannot be less than and more than 0 at the same time, so I would use elseif to avoid evaluating both conditions if not necessary. But if the above works, I would rather use that.